README.md

```diff
@@ -1,6 +1,69 @@
 # External Health Monitor
 
-This repo contains sidecar controller and agent for volume health monitoring.
+The External Health Monitor is part of Kubernetes implementation of [Container Storage Interface (CSI)](https://github.com/container-storage-interface/spec). It was introduced as an Alpha feature in Kubernetes v1.19.
+
+## Overview
+
+The [External Health Monitor](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1432-volume-health-monitor) is impelmeted as two components: `External Health Monitor Controller` and `External Health Monitor Agent`.
+
+- External Health Monitor Controller:
+  - The external health monitor controller will be deployed as a sidecar together with the CSI controller driver, similar to how the external-provisioner sidecar is deployed.
+  - Trigger controller RPC to check the health condition of the CSI volumes.
+  - The external controller sidecar will also watch for node failure events. This component can be enabled via a flag.
+
+- External Health Monitor Agent:
+  - The external health monitor agent will be deployed as a sidecar together with the CSI node driver on every Kubernetes worker node.
+  - Trigger node RPC to check volume's mounting conditions.
+  - Note that currently we do not have CSI support for local storage. When the support is available, we will implement relavant CSI monitoring interfaces as well.
+
+The External Health Monitor needs to invoke the following CSI interface to support
+
+- External Health Monitor Controller:
+  - NodeGetVolumeStats
+  - ControllerGetVolume
+- External Health Monitor Agent:
+  - ListVolumes
+
+## Usage
+
+Because External Health Monitor needs controller and agent simultaneously, you have to deploy them to your Kubernetes cluster.
+
+### External Health Monitor Controller
+
+```bash
+cd external-health-monitor
+kubectl create -f deploy/kubernetes/external-health-monitor-controller
+```
+
+### External Health Monitor Agent
+
+```bash
+kubectl create -f deploy/kubernetes/external-health-monitor-agent
+```
+
+You can run `kubectl get pods` command to confirm if they are deployed on your cluster successfully.
+
+```bash
+> kubectl get pods -o wide
```

Comment:
Remove these print outs as they used your private image.  This should be added back after the release is cut and official image becomes available.